pkg/kp/rcstore/consul_store.go

```diff
@@ -451,8 +451,32 @@ func (s *consulStore) SetDesiredReplicas(id fields.ID, n int) error {
 }
 
 func (s *consulStore) AddDesiredReplicas(id fields.ID, n int) error {
```

Comment:
alternatively, maybe the expected initial value can be passed in by the roll algorithm - after all, [it knows (`oldNodes.Desired` and `newNodes.Desired`)]https://github.com/square/p2/blob/master/pkg/roll/run_update.go#L248-L267